### PR TITLE
Location event + edit event

### LINKED ISSUE #36: Choose and add a Location to an event + Edit Event
Add a drop down menu to choose a location for an event

### PR DESCRIPTION
**Pull request by SamBachmann**
In this PR, I refactored the code of CreateEvent to use it for EditEvent. I implemented the GUI and functionalites to edit an event. I add the categories when creating an event and a way to choose a location for an event.

closes #36

### CODE REVIEW

File: app/src/main/java/com/github/se/gatherspot/model/EventUtils.kt
Comment (Eliasgoat) [inline app/src/main/java/com/github/se/gatherspot/model/EventUtils.kt:1]: It is nice that we change the name of this class.

File: app/src/main/java/com/github/se/gatherspot/ui/EventDataForm.kt
Comment (Eliasgoat) [inline app/src/main/java/com/github/se/gatherspot/ui/EventDataForm.kt:1]: Very good modularity

**Review by Eliasgoat — COMMENTED** [review]
The code looks good on my end. Very good modularity. We just have to make sure that the correct eventFirebase test is in the main from PR #47

File: app/src/main/java/com/github/se/gatherspot/ui/EventDataForm.kt
Comment (markh0rr) [reply app/src/main/java/com/github/se/gatherspot/ui/EventDataForm.kt:1]: Good Testing habits attaining a good coverage

**Comment by markh0rr** [comment]
Good Testing habits attaining a good coverage

File: app/src/main/java/com/github/se/gatherspot/ui/EventDataForm.kt
Comment (markh0rr) [reply app/src/main/java/com/github/se/gatherspot/ui/EventDataForm.kt:1]: Good Testing habits attaining a good coverage

**Review by markh0rr — APPROVED** [review]
Good Testing habits attaining a good coverage

**Pull request merged**